package/MDAnalysis/core/universe.py

```diff
@@ -113,20 +112,38 @@ def _topology_from_file_like(topology_file, topology_format=None,
         # one because the file isn't present
         # or the permissions are bad, second when the parser fails
         if (err.errno is not None and
-            errno.errorcode[err.errno] in ['ENOENT', 'EACCES']):
-            # Runs if the error is propagated due to no permission / file not found
+                errno.errorcode[err.errno] in ['ENOENT', 'EACCES']):
+            # Runs if the error is propagated due to no permission / file not
+            # found
             raise sys.exc_info()[1] from err
         else:
             # Runs when the parser fails
             raise IOError("Failed to load from the topology file {0}"
-                            " with parser {1}.\n"
-                            "Error: {2}".format(topology_file, parser, err))
+                          " with parser {1}.\n"
+                          "Error: {2}".format(topology_file, parser, err))
     except (ValueError, NotImplementedError) as err:
         raise ValueError(
             "Failed to construct topology from file {0}"
             " with parser {1}.\n"
             "Error: {2}".format(topology_file, parser, err))
-    return topology
+    txyz = False
+    begin_guess = False
+    if hasattr(parser, 'format'):
+        formats = []
+        if not isinstance(parser.format, list):
+            formats.append(parser.format)
+        else:
+            formats = parser.format
+        if ('MINIMAL' not in formats and 'TPR' not in formats and
+            'THINGY' not in formats and topology.n_atoms > 0 and
+                any(fmt in _PARSERS for fmt in formats)):
+            begin_guess = True
+        # check if a file is txyz format to handle its
+        # to preserve its special behavior of guessing
+        # atom masses from names
+        if 'TXYZ' in formats or 'ARC' in formats:
+            txyz = True
+    return [topology, begin_guess, txyz]
```

Comment:
What is the meaning of the begin_guess variable?

Reply:
> About [#3753 (comment)](https://github.com/MDAnalysis/mdanalysis/pull/3753#issuecomment-1200827768)
> 
> > 
> 
> I have seen many changes recently, but none seemed to touch the line I mentioned above. Just to clarify, I expect that the issue with Universe.empty can be solved by changing this line alone.
> 
> Universe.empty is a "class method". Instead of taking an instance of the class, `self`, as first argument, it takes the class itself as `cls`. The line I am talking about creates the universe, it is equivalent to `mda.Universe(top)` therefore you can pass all the arguments a universe can take. Including the ones about guessing. I expect you can ask there for nothing to be guessed.

the problem is even if you created an empty universe, there is an automatic call for mass and type guessing inside the universe, which causes errors. This problem also happens with many other cases as reading topology from a coordinate file or NumPy array . The point of this automatic guessing is to mimic the default behavior of the topology parsers, so I'm trying to not trigger it except in the case of reading topology from a typical topology file, this is the idea behind `begin_guess flag`

Reply:
There should not be that call. One of the reason we do all of these changes is to be able to not guess anything.

I had a look at why you have that call. You should be able to get rid of it by adjust the default value of the `to_guess` argument as I suggested earlier.

Reply:
This has the same result indeed... defualt value for to_guess tuple will not prevent automatic guessing from happening when it is not intended to be (empty universe, etc.). That's why I'm trying to check that guessing happens when a topology came from a parser that already used to call guessers, not any other source that doesn't need any automatic guessing to happen.

I don't remember why I removed the default values from the to_guess list. Most probably because not all parsers guess masses and types, so we must check first befor adding them to the to_guess list by defualt.

All of this redundant checkpoints is just for not breaking the defualt behaviour of the current version of the library. Once it's official that no guessing should takes place without the user intention, all of this will have no functionality and should be removed.